app/src/main/java/com/gholem/moneylab/features/add/AddTransactionFragment.kt

```diff
@@ -30,12 +32,22 @@ class AddTransactionFragment : BaseFragment<FragmentAddBinding, AddTransactionVi
         observeActions()
         viewModel.init()
 
+        val navController = findNavController();
```

Comment:
This method is getting long so you can exclude some code to separate method and just trigger it here:

```
private fun observeCategoryChange() {
        findNavController().currentBackStackEntry?.savedStateHandle?.getLiveData<Int>(KEY_CATEGORY)
            ?.observe(
                viewLifecycleOwner
            ) { result ->
                dataAdapter.setCategory(result)
            }
}
```